swiftlang/swift-docc
# Revert "Revert "Define commands as asynchronous and use Task for preview cancellation ""

Reverts swiftlang/swift-docc#1058

## Testing

To reproduce the crash caused by rdar://137774949 you need to run the tests in a release configuration with a recent Swift compiler. For me it reproduces with `swiftlang-6.1.0.1.51` but it's possible that you can go further back than that.

- Checkout https://github.com/swiftlang/swift-docc/pull/1062/commits/a9f1c0b8a8888aee12ecfb0e4d1e8e5398ba30fd
- Run `swift test --parallel --configuration release --filter SwiftDocCUtilitiesTests.PreviewActionIntegrationTest` 
  - Confirm that it reproduces the crash. If it doesn't the the steps below won't verify anything.
- Checkout https://github.com/swiftlang/swift-docc/pull/1062/commits/2eafe4b187dea498f62c8a177269feecc7c1555b 
- Run `for i in {1..10}; do swift test --parallel --configuration release --filter SwiftDocCUtilitiesTests.PreviewActionIntegrationTests; done`
  - The tests should repeatedly pass
